openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/schematypes/ArraySchemaType.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.openapi.generators.schema.schematypes;
+
+import io.ballerina.compiler.syntax.tree.TypeDescriptorNode;
+import io.ballerina.openapi.exception.BallerinaOpenApiException;
+import io.ballerina.openapi.generators.schema.SchemaUtils;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.Schema;
+
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createIdentifierToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createSimpleNameReferenceNode;
+import static io.ballerina.openapi.generators.GeneratorConstants.MAX_ARRAY_LENGTH;
+import static io.ballerina.openapi.generators.GeneratorConstants.NILLABLE;
+
+/**
+ * Generate TypeDefinitionNode and TypeDescriptorNode for array schemas.
+ *
+ * @since 2.0.0
+ */
+public class ArraySchemaType extends SchemaType {
+    private final OpenAPI openAPI;
+    private final boolean nullable;
+
+    public ArraySchemaType(OpenAPI openAPI, boolean nullable) {
+        this.openAPI = openAPI;
+        this.nullable = nullable;
+    }
+
+    /**
+     * Generate TypeDescriptorNode for array type schemas. If array type is not given, type will be `AnyData`
+     */
+    @Override
+    public TypeDescriptorNode generateTypeDescriptorNode(Schema schema) throws BallerinaOpenApiException {
+        assert schema instanceof ArraySchema;
+        ArraySchema arraySchema = (ArraySchema) schema;
+        String fieldTypeNameStr;
+        fieldTypeNameStr = SchemaUtils.getBallerinaTypeName
+                (arraySchema.getItems(), this.nullable, this.openAPI).toString().trim();
```

Comment:
@indikasampath2000 @abeykoon 
I am afraid that we have to create relevant `SchemaType` object via `SchemaTypeFactory` again  to figure out the type of the array, because array items can belong to
- primitive schema
- referenced schema
- composed schema
- arrayschema
- object schema - handled by flatten 

to check this and generate accordingly we again need to have an if-else statement. 

In the current way it'll even support the below scenario generating `string[][][]tag?;` like this. 

```yaml
tag:
          type: array
          items:
            type: array
            items:
              type: array
              items:
                type: string
```